src/lib/pmd/PmdWrapper.ts

```diff
@@ -104,6 +113,8 @@ export default class PmdWrapper extends PmdSupport {
 
 		cp.on('exit', code => {
 			this.logger.trace(`monitorChildProcess has received exit code ${code}`);
+			// Cleanup the temp file
```

Comment:
This is still synchronous because this method is called from within a promise and the method itself isn't async. I think that this is OK.